### DISCUSSION TITLE
Recording in mono

### DISCUSSION THREAD

**Discussion by coulisse** [pschatzmann/arduino-audiokit#13] [Q&A] [answered]
Hello
is it possible with **arduino-audiokit**  to record in mono mode?

I see that in config there is not the possiblity to specify the _number of channels_; there is only the **fmt** property  that could be valued with:
    I2S_CHANNEL_FMT_RIGHT_LEFT,   I2S_CHANNEL_FMT_ALL_RIGHT,     I2S_CHANNEL_FMT_ALL_LEFT,
    I2S_CHANNEL_FMT_ONLY_RIGHT,   I2S_CHANNEL_FMT_ONLY_LEFT.

Thanks in advance.
Corrado.

**Comment by pschatzmann** [comment, marked as answer]
It is unclear to me what you try to do, but I would expect that cfg.channels = 1; might be what you are looking for...

**Reply by coulisse** [reply]
Thanks for your always quick replay!
I would receive in mono instead of stereo from i2s.

If i referr to you example:
[https://github.com/pschatzmann/arduino-audiokit/blob/main/examples/input/input.ino](https://github.com/pschatzmann/arduino-audiokit/blob/main/examples/input/input.ino)
is not possible to specify cfg.channesl=1

I got the following error:

`error: 'struct AudioKitConfig' has no member named 'channels'`


Thanks
Corrado

**Reply by coulisse** [reply]
In addition to previous question, i tell you that I made a trick and now it is running in mono!  :-)
But I think this is not the proper way...

After begin i call **i2s_set_clk**:
```
kit.begin(cfg);
i2s_set_clk(I2S_NUM_0,11025,I2S_BITS_PER_SAMPLE_16BIT,I2S_CHANNEL_MONO); 
```

Best regards
Corrado

**Comment by pschatzmann** [comment]
I recommend to use the functionality from the arduino-audio-tools. It should work from there ...

I am not sure what exectly the impact of i2s_set_clk() is which is impacting only the ESP32 I2S side, but the sound chip is still producing and consuming stereo data.
